sections/main-product.liquid

```diff
@@ -49,23 +49,36 @@
       {% if product_system != nil %}› <a href="/collections/{{ product_system | downcase | replace: "/", "" | replace: "  ", " " | replace: " ", "-" }}" class="link">{{ product_system }}</a>{% endif %}
     </div>
   {% endif %}
-  <div class="product grid grid--1-col {% if product.media.size > 0 %}grid--2-col-tablet{% else %}product--no-media{% endif %}">
+  <div class="product grid grid--1-col grid--2-col-tablet">
```

Comment:
I'm guessing that you removed this because now there should always be an image.
Is there no possible case at all, even through some other bug for example, where the product ends up still not having any image?

Reply:
Solved, now it checks also if the host product has images and an if not, then adds the class "product--no-media"